projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart.e2e-spec.ts

```diff
@@ -73,4 +61,203 @@ describe('Cart', () => {
       cy.get('.cart-details-wrapper .cx-total').contains(`Cart #${cartCode}`);
     });
   });
+
+  it('should be loaded after user login', () => {
+    cy.server();
+    cart.registerCartUser();
+    cart.loginCartUser();
+    cy.visit(`/product/${cart.products[0].code}`);
+    cart.addToCart();
+    cart.checkAddedToCartDialog();
+    cart.closeAddedToCartDialog();
+    cy.selectUserMenuOption({
+      option: 'Sign Out',
+    });
+    cy.clearLocalStorage();
+    cy.route(
+      `${apiUrl}/rest/v2/electronics-spa/users/current/carts?fields=*`
+    ).as('carts');
+    cart.loginCartUser();
+    cy.wait('@carts');
+    cy.visit('/cart');
+    cart.checkProductInCart(cart.products[0]);
+
+    // cleanup
+    cart.removeCartItem(cart.products[0]);
+    cart.validateEmptyCart();
+  });
+
+  it('should load cart saved in browser storage', () => {
+    cy.server();
+    cart.loginCartUser();
+    cy.visit(`/product/${cart.products[0].code}`);
+    cart.addToCart();
+    cart.checkAddedToCartDialog();
+    cart.closeAddedToCartDialog();
+    cy.reload();
+    cy.route(`${apiUrl}/rest/v2/electronics-spa/users/current/carts/*`).as(
+      'cart'
+    );
+    cy.wait('@cart');
+    cy.visit('/cart');
+    cart.checkProductInCart(cart.products[0]);
+
+    // cleanup
+    cart.removeCartItem(cart.products[0]);
+    cart.validateEmptyCart();
+  });
+
+  context('add entry', () => {
+    // will fail right now, as this is not implemented yet
+    it.skip('should first try to load cart when adding first entry for logged user', () => {
+      cy.server();
+      login(
+        cart.cartUser.registrationData.email,
+        cart.cartUser.registrationData.password,
+        false
+      ).then(res => {
```

Comment:
Small improvements/nitpicks not required for the final version mark with `nit:` prefix.

Those can be fixed, but are not important.